lib/galaxy/tool_util/deps/container_resolvers/explicit.py

```diff
@@ -77,11 +78,10 @@ def resolve(self, enabled_container_types, tool_info, install=False, **kwds):
         hence the container_description hack here.
         """
         for container_description in tool_info.container_descriptions:  # type: ContainerDescription
+            container_description = copy.copy(container_description)
             if container_description.type == 'docker':
-                desc_dict = container_description.to_dict()
-                desc_dict['type'] = self.container_type
-                desc_dict['identifier'] = f"docker://{container_description.identifier}"
-                container_description = container_description.from_dict(desc_dict)
+                container_description.type = self.container_type
+                container_description.identifier = f"docker://{container_description.identifier}"
```

Comment:
Isn't that exactly what the original code does ?
`desc_dict = container_description.to_dict()` gives you a new dict, so setting dict keys and values doesn't do anything to the original ContainerDescription instance, and then in
`container_description = container_description.from_dict(desc_dict)` you're building a new instance using `ContainerDescription.from_dict`, which also doesn't modify the source.